File: show_middle.py
"""def get_middle(s):
    return s[(len(s)-1)//2: len(s)//2+1]
    """


def get_middle(s):
    size = len(s)
    half = size//2
    isEven = size%2
     
    if isEven == 0:
       return s[half-1 : half+1]
    else:
       return s[half]
        
print(get_middle("words"))

File: ShowMiddle.java
public class ShowMiddle {
    public static void main(String[] args) {
        System.out.println(getMiddle("sentences"));
    }

    public static String getMiddle(String sentence) {
        int size = sentence.length();
        int half = size / 2;
        int isEven = size % 2;

        if (isEven == 0) {
            return sentence.substring(half - 1, half + 1);
        } else {
            return Character.toString(sentence.charAt(half));
        }
    }
}
